File: ruby/count_repeats.rb
# https://www.codewars.com/kata/598ee7b6ec6cb90dd6000061/train/ruby

def count_repeats(text)
  return 0 if text.chars == text.chars.uniq
  count = 0
  
  text.chars.each.with_index { |letter, index| count += 1 if letter == text[index + 1] }  
  count
end

File: javascript/countRepeats.js
function countRepeats(text) {
    if (new Set(text).size !== text.length) {
      let count = 0;
      for (let i = 0; i < text.length - 1; i++) {
        if (text[i] === text[i + 1]) {
          count++;
        }
      }
      return count;
    }
    return 0;
  }